File: languages/kotlin/src/model/SimpleTableAnalysis.kt
package model

import box.UserInfo

// Criando uma interface "SimpleTableAnalysis"
interface SimpleTableAnalysis {

    /**
     * Método de interface, responsável por realizar a análise na lista de "UserInfo"
     * userInfoList: List<UserInfo> -> Lista de "UserInfo"
     * RETURN: Double -> valor da análise
     */
    fun analysis(userInfoList: List<UserInfo>): Double
}

File: languages/typescript/src/model/SimpleTableAnalysis.ts
// Importando bibliotecas
import {UserInfo} from './UserInfo'

// Criando uma interface "SimpleTableAnalysis"
export interface SimpleTableAnalysis {
    // Método de interface, responsável por realizar a análise na lista de "UserInfo"
    // list: List<UserInfo> -> Lista de "UserInfo"
    // RETURN: double -> valor da análise
    analysis(userInfoList: Array<UserInfo>): number
}